mantidproject/mantid
# Add option get TOF window from back-to-back exp parameters in IntegratePeaksSkew

**Description of work.**

Add option get TOF window from back-to-back exponential parameters in `IntegratePeaksSkew` - this is primarily for WISH as the optimal TOF window used for the initial integration does not agree well with the TOF resolution equation (because the width is not dominated by resolution but by the tails of the back-to-back exponential).

Also simplified the algorithm input such that if the user wants to integrate with an initial window dTOF/TOF = constant for all peaks then this can be achieved by setting `ThetaWidth = 0` rather than a separate input parameter  -this is technically a breaking change, but the algorithm is only in use by WISH scientists atm.

I have also added some  back-to-back exponential parameters for SXD to the parameters.xml file - this will let users on SXD to get an estimate for the appropriate resolution parameters in `IntegratePeaksSkew` without needing an initial guess. 
The value for `A` is hard coded (very steep, so width to low TOF is dominated by gaussian sigma, `S`) and the coefficients for `B` are taken from SXD2001 source code (written by Matthias Gutmann) - it is a function of wavelength not d-spacing so can be applied to the entire instrument rather than a specific bank. The coefficients for `S` are taken from WISH 90 degree banks which seem to perform OK  (this is a coincidence, there is no reason for them to be the same - technically SXD should have coeficients for each bank as the dTOF/TOF resolution  depends on scattering angle).

**To test:**
(1) Run this script
```
Load(Filename=r'C:\mantid\build\ExternalData\Testing\Data\SystemTest\SXD23767.raw', OutputWorkspace='SXD23767')
CreatePeaksWorkspace(InstrumentWorkspace='SXD23767', NumberOfPeaks=0, OutputWorkspace='SingleCrystalPeakTable')

AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=3271, DetectorID=32615)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=3806, DetectorID=30017)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=5009, DetectorID=42513)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=2486, DetectorID=39170)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=2613, DetectorID=40740)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=2328, DetectorID=40905)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=1703, DetectorID=28764)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=2449, DetectorID=41068)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=2303, DetectorID=40121)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=2038, DetectorID=36910)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=3787, DetectorID=42749)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=886, DetectorID=14284)
AddPeak(PeaksWorkspace='SingleCrystalPeakTable', RunWorkspace='SXD23767', TOF=1817, DetectorID=36104)
  
out = IntegratePeaksSkew(InputWorkspace='SXD23767', PeaksWorkspace='SingleCrystalPeakTable',
    OutputFile="out.pdf", OutputWorkspace='out',  UseNearestPeak=True, IntegrateIfOnEdge=True, 
    NVacanciesMax=5, NPixPerVacancyMin=1, UpdatePeakPosition=True, OptimiseMask=True, 
    ScaleThetaWidthByWavelength=True, GetTOFWindowFromBackToBackParams=True)
```
(2) Check in the `out.pdf` file that the initial TOF window looks OK

Fixes #34562 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
